### PR TITLE
Shorten numpy

### PR DESCRIPTION
**Pull request by mcocdawc**
shortened numpy calls and replaced `functools.reduce(numpy.dot(...` with `multi_dot`;
it's easier to read and faster.

### CODE REVIEW

**Comment by mscho527** [comment]
Before I review this, do we want `multidot((A, B, C))` or `A @ B @ C`?
Either way, we should probably add this to [https://github.com/troyvvgroup/quemb/wiki/Preferred-Operations](https://github.com/troyvvgroup/quemb/wiki/Preferred-Operations) as we merge this PR.

File: src/quemb/kbe/lo_k.py [hunk 1]
```diff
@@ -125,26 +114,24 @@ def get_iao_k(Co, S12, S1, S2=None, ortho=True):
...
-            - numpy.dot(
-                (Po + Potil - 2.0 * reduce(numpy.dot, (Po, S1[k], Potil))), S1[k]
-            )
+            - numpy.dot((Po + Potil - 2.0 * multi_dot((Po, S1[k], Potil))), S1[k])
         ) @ ptil
         if ortho:
             Ciao[k] = symm_orth_k(Ciao[k], ovlp=S1[k])
 
-            rep_err = numpy.linalg.norm(Ciao[k] @ Ciao[k].conj().T @ S1[k] @ Po - Po)
+            rep_err = norm(Ciao[k] @ Ciao[k].conj().T @ S1[k] @ Po - Po)
```
Comment (mscho527) [inline src/quemb/kbe/lo_k.py:134]: `multi_dot` here as well?

**Comment by mcocdawc** [comment]
> Before I review this, do we want `multidot((A, B, C))` or `A @ B @ C`? Either way, we should probably add this to https://github.com/troyvvgroup/quemb/wiki/Preferred-Operations as we merge this PR.

Adjusted the wiki
https://github.com/troyvvgroup/quemb/wiki/Preferred-Operations
per our discussion @mscho527 .

File: src/quemb/kbe/lo_k.py [hunk 2]
```diff
@@ -47,17 +41,12 @@ def get_symm_orth_mat_k(A, thr=1.0e-7, ovlp=None):
             "smallest eigenvalue (%.3E) is less than thr (%.3E). "
             "Please use 'cano_orth' instead." % (numpy.min(e), thr)
         )
-    U = reduce(numpy.dot, (u, numpy.diag(e**-0.5), u.conj().T))
-    # U = reduce(numpy.dot, (u/numpy.sqrt(e), u.conj().T))
-    return U
+    return u @ numpy.diag(e**-0.5) @ u.conj().T
```
[the remark this answers is not in the document]
Comment (mscho527) [reply src/quemb/kbe/lo_k.py:44]: oh yes sure -- my bad!

**Review by mscho527 — APPROVED** [review]
```
                __                           __
 _______  ___  / / ___ ___    _______  ___  / /
/ __/ _ \/ _ \/ / (_-</ _ \  / __/ _ \/ _ \/ / 
\__/\___/\___/_/ /___/\___/  \__/\___/\___/_/  
                                               
```

**Pull request merged**